lldb/source/Plugins/Language/Swift/SwiftLanguage.cpp

```diff
@@ -1894,6 +1894,219 @@ SwiftLanguage::GetDemangledFunctionNameWithoutArguments(Mangled mangled) const {
   return mangled_name;
 }
 
+static std::optional<llvm::StringRef>
```

Comment:
an optional StringRef is redundant unless there is a semantic difference between std::null_opt and an empty string.

Reply:
I would just return a StringRef here

Reply:
FWIW I did this in the upstream C++ plugin too. I did this because this way we can differentiate between "failed to extract this demangling info" versus "this part of the demangled string was not present, but nothing failed". In the failure case we return `false` from the frame-format variable handler, which is necessary to trigger the fallback format in case things went wrong.

I guess instead of optional I should've probably used `Expected` and logged an error. Might be a good improvement for upstream actually if @charles-zablit has time on his hands :)

Reply:
Opened a PR upstream to address this:
- https://github.com/llvm/llvm-project/pull/144731

Reply:
Thanks, update the functions to use `llvm::Expected`.

There seems to be a lot of duplicated code between the CPP and Swift plugin now. The 5 or so functions are pretty much the same. I think we could (in a follow up patch) implement a class that can be inherited and which defines these methods? I'm not sure if that's common practice.